graylog2-server/src/main/java/org/graylog/events/processor/systemnotification/SystemNotificationEventProcessorConfig.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.events.processor.systemnotification;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+import com.google.common.graph.MutableGraph;
+import org.graylog.events.contentpack.entities.EventProcessorConfigEntity;
+import org.graylog.events.processor.EventProcessorConfig;
+import org.graylog2.contentpacks.EntityDescriptorIds;
+import org.graylog2.contentpacks.model.entities.EntityDescriptor;
+import org.graylog2.plugin.rest.ValidationResult;
+
+@AutoValue
+@JsonTypeName(SystemNotificationEventProcessorConfig.TYPE_NAME)
+@JsonDeserialize(builder = SystemNotificationEventProcessorConfig.Builder.class)
+public abstract class SystemNotificationEventProcessorConfig implements EventProcessorConfig {
+    public static final String TYPE_NAME = "system-notifications-v1";
+
+    public static Builder builder() {
+        return Builder.create();
+    }
+
+    public abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    public abstract static class Builder implements EventProcessorConfig.Builder<Builder> {
+        @JsonCreator
+        public static Builder create() {
+            return new AutoValue_SystemNotificationEventProcessorConfig.Builder()
+                    .type(TYPE_NAME);
+        }
+
+        public abstract SystemNotificationEventProcessorConfig build();
+    }
+
+    // TODO: Implement
+    @Override
+    public ValidationResult validate() {
+        return new ValidationResult();
+    }
+
+    @Override
+    public EventProcessorConfigEntity toContentPackEntity(EntityDescriptorIds entityDescriptorIds) {
+        // TODO: Don't forget to add the correct Jackson subtype mapping, so that the content pack import is successful:
+        //  e.g. registerJacksonSubtype(SigmaEventProcessorConfigEntity.class,
+        //                SigmaEventProcessorConfigEntity.TYPE_NAME);
```

Comment:
```suggestion
    // Don't export this into content packs
```

Reply:
Hmm, on a second thought, we might want to export them, but we need some special handling,
because the entity itself should always exist, we would only update the notification settings via content packs...

Reply:
Per offline discussion we won't export this into content packs at this time.